### DISCUSSION TITLE
IOS

### DISCUSSION THREAD

**Discussion by Sampyuays** [geode-sdk/sdk#548] [Help] [answered]
Is it ever going to be added to IPhone? thought i would ask as the previous message has not got a response

**Comment by Djspac3** [comment]
**BASED ON WHAT I HAVE HEARD**

Probably never it's apple you need to make fresh code and compile it with "xcode" too or what ever

 Edit oh there mac

**Reply by TheSillyDoggo** [reply]
It's not possible without jailbreak because Ios doesn't let you inject dylibs at runtime

**Reply by Djspac3** [reply]
Oh, bruh, atliest it's one more reason to stay with andriod 👍🏻

**Reply by Sampyuays** [reply]
but waht about ICreate, that works without jailbreak or app store

**Reply by TheSillyDoggo** [reply]
I create is different

**Reply by RoootTheFox** [reply]
iCreate is a static, one time patch to the game and doesn't load mods at runtime. Geode however, being a mod *loader*, DOES load mods at runtime - which iOS doesn't allow.

**Reply by RoootTheFox** [reply]
tldr: iCreate works *very* differently from Geode, and the way Geode works (and has to work) isn't possible on iOS

(at least not without a jailbreak or very annoying and user+developer unfriendly workarounds)

**Reply by heavyutils** [reply]
> iCreate is a static, one time patch to the game and doesn't load mods at runtime. Geode however, being a mod *loader*, DOES load mods at runtime - which iOS doesn't allow.

Does it? If so, why do I need to restart my game everytime I want to apply mods?

**Reply by RoootTheFox** [reply]
> > iCreate is a static, one time patch to the game and doesn't load mods at runtime. Geode however, being a mod _loader_, DOES load mods at runtime - which iOS doesn't allow.
> 
> Does it? If so, why do I need to restart my game everytime I want to apply mods?

This is an intentional limitation of Geode that was put in place to avoid undefined behaviour (read: crashes)

**Reply by RoootTheFox** [reply]
What iCreate does is patch the GD binary directly, making a modded game as its own app
Geode **can not** do that because it's a mod **loader** - Geode **injects** both itself and mods **at runtime** (when you open GD) instead of modifying GD itself, and all the patching happens from that point

iOS does NOT allow **runtime** modification of code¹, which Geode **has to do**, but iCreate doesn't - that's why iCreate works and Geode doesn't 

<sub>¹ it is possible but not without either a jailbreak (which may cause people to brick their devices trying to find a non-existant jailbreak) or extremely user-unfriendly workarounds that involve a computer every time you want to open GD</sub>

**Comment by bp88dev** [comment]
Long story short from what I've heard: iOS is a pain in the ass to compile for. Probably not.

**Reply by RoootTheFox** [reply]
that's not the issue. the issue is getting it to *run* on iOS, given all of its restrictions.

**Comment by antonfdiaz** [comment]
I don't think it's gonna be available for iOS, at least without jailbreak, because it requires injecting dylib's into the game and so on. Also for compiling you need Xcode and probably it's a pain to use, and it can take up more than 20gb on your Mac. But it would be cool to have it on iOS :)

**Reply by FireMario211** [reply]
this aged badly

**Reply by RoootTheFox** [reply]
@FireMario211 especially because this was posted around the time the ios launcher repo went public and less than a month before the public release ^-^

**Comment by Hbg1010** [comment, marked as answer]
Facts
![IMG_2022](https://github.com/user-attachments/assets/2ca240e4-2db8-4413-b42f-f1bed560756c)

**Reply by antonfdiaz** [reply]
oh, ok... but how?

**Reply by iNUCi** [reply]
did you not notice https://github.com/geode-sdk/ios-launcher

**Reply by antonfdiaz** [reply]
thanks i didn't know that!
